frameit/lib/frameit/device.rb

```diff
@@ -51,7 +51,7 @@ def self.detect_device(path, platform)
       found_device = nil
       filename_device = nil
       filename = Pathname.new(path).basename.to_s
-      Devices.constants.each do |c|
+      Devices.constants.sort_by(&:length).reverse_each do |c|
```

Comment:
This is a crazy assumption here. But quite safe, IMO. It doesn't rely on business logic or marketing naming standards, but rather "the longer the string, the more specific it will be" and thus matching with partial substrings won't be possible. Quite clever IMO :)